src/detect-base64-decode.c

```diff
@@ -82,24 +77,15 @@ int DetectBase64DecodeDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s
 
     decode_len = MIN(payload_len, data->bytes);
 
-#if 0
-    printf("Decoding:\n");
-    PrintRawDataFp(stdout, payload, decode_len);
-#endif
-
     uint32_t consumed = 0, num_decoded = 0;
-    (void)DecodeBase64(det_ctx->base64_decoded, det_ctx->base64_decoded_len_max, payload,
-            decode_len, &consumed, &num_decoded, BASE64_MODE_RELAX);
+    Base64Ecode code = DecodeBase64(det_ctx->base64_decoded, det_ctx->base64_decoded_len_max,
+            payload, decode_len, &consumed, &num_decoded, BASE64_MODE_RELAX);
+    if (code == BASE64_ECODE_ERR) {
+        return 0;
```

Comment:
pretty sure this is incorrect. We want to inspect available data even if at some point there is a data error

anyway, I believe the coverity error is already resolved in master by the `(void)` cast

Reply:
Confirmed by rerunning coverity on master (last run was on a dev branch)

Reply:
failed to notice the relaxed mode.

Reply:
Oh...

How come coverity run on a dev branch ? Do you trigger builds manually ?

Reply:
yes all builds are triggered manually by me

Reply:
Ok thanks for the info, will check the branch/commit next time ;-)

(I thought it was automatic on master branch)